bundles/model/org.eclipse.smarthome.model.item/src/org/eclipse/smarthome/model/item/internal/GenericItemProvider.java

```diff
@@ -400,8 +402,76 @@ private boolean hasItemChanged(Item item1, Item item2) {
                 !Objects.equals(item1.getGroupNames(), item2.getGroupNames()) || //
                 !Objects.equals(item1.getLabel(), item2.getLabel()) || //
                 !Objects.equals(item1.getTags(), item2.getTags()) || //
-                !Objects.equals(item1.getType(), item2.getType());
+                !Objects.equals(item1.getType(), item2.getType()) || //
+                hasStateDescriptionChanged(item1.getName(), item2.getName()) || //
+                hasGroupItemChanged(item1, item2);
+    }
+
+    private boolean hasStateDescriptionChanged(String itemName1, String itemName2) {
+        StateDescription sd1 = stateDescriptions.get(itemName1);
+        StateDescription sd2 = stateDescriptions.get(itemName2);
+
+        if (sd1 == null && sd2 == null) {
+            return false;
+        }
+
+        else if ((sd1 != null && sd2 == null) || (sd1 == null && sd2 != null)) {
+            // A StateDescription was added or removed
+            return true;
+        }
+
+        return !Objects.equals(sd1.getPattern(), sd2.getPattern());
+    }
+
+    private boolean hasGroupItemChanged(Item item1, Item item2) {
+        GroupItem gItem1 = null;
+        GroupItem gItem2 = null;
+
+        if (item1 instanceof GroupItem) {
+            gItem1 = (GroupItem) item1;
+        }
+        if (item2 instanceof GroupItem) {
+            gItem2 = (GroupItem) item2;
+        }
+
+        if (gItem1 == null && gItem2 == null) {
+            return false;
+        }
+
+        if ((gItem1 != null && gItem2 == null) || (gItem1 == null && gItem2 != null)) {
+            // Item was changed into a GroupItem
+            return true;
+        }
+
+        boolean sameBaseItemClass = false;
+        if (gItem1.getBaseItem() != null && gItem2.getBaseItem() != null) {
+            sameBaseItemClass = Objects.equals(gItem1.getBaseItem().getClass(), gItem2.getBaseItem().getClass());
+        }
+
+        boolean sameFunction = false;
+        GroupFunction gf1 = gItem1.getFunction();
+        GroupFunction gf2 = gItem2.getFunction();
+        if (gf1 != null && gf2 != null) {
```

Comment:
see above

Reply:
I have added an `else` branch to cover the "double null" case.